econml/drlearner.py

```diff
@@ -0,0 +1,633 @@
+                return np.hstack([arr for arr in [X, W] if arr is not None])
+
+            def fit(self, Y, T, X=None, W=None, *, sample_weight=None):
+                assert np.ndim(Y) == 1, "Can only accept single dimensional outcomes Y! Use Y.ravel()."
+                if (X is None) and (W is None):
+                    raise AttributeError("At least one of X or W has to not be None!")
+                if np.any(np.all(T == 0, axis=0)) or (not np.any(np.all(T == 0, axis=1))):
+                    raise AttributeError("Provided crossfit folds contain training splits that " +
+                                         "don't contain all treatments")
+                XW = self._combine(X, W)
+                filtered_kwargs = _filter_none_kwargs(sample_weight=sample_weight)
+                self._model_propensity.fit(XW, np.matmul(T, np.arange(1, T.shape[1] + 1)).ravel().astype(int),
+                                           **filtered_kwargs)
+                self._model_regression.fit(np.hstack([XW, T]), Y, **filtered_kwargs)
+                return self
+
+            def predict(self, Y, T, X=None, W=None, *, sample_weight=None):
+                XW = self._combine(X, W)
+                propensities = self._model_propensity.predict_proba(XW)
+                Y_pred = np.zeros((T.shape[0], T.shape[1] + 1))
+                T_counter = np.zeros(T.shape)
+                Y_pred[:, 0] = self._model_regression.predict(np.hstack([XW, T_counter]))
+                Y_pred[:, 0] += (Y - Y_pred[:, 0]) * np.all(T == 0, axis=1) / propensities[:, 0]
+                for t in np.arange(T.shape[1]):
+                    T_counter = np.zeros(T.shape)
+                    T_counter[:, t] = 1
+                    Y_pred[:, t + 1] = self._model_regression.predict(np.hstack([XW, T_counter]))
+                    Y_pred[:, t + 1] += (Y - Y_pred[:, t + 1]) * (T[:, t] == 1) / propensities[:, t + 1]
+                return Y_pred
+
+        class ModelFinal:
+            def __init__(self, model_final, featurizer, multitask_model_final):
+                self._model_final = clone(model_final, safe=False)
+                self._featurizer = clone(featurizer, safe=False)
+                self._multitask_model_final = multitask_model_final
+                return
+
+            def _filter_none_kwargs(self, **kwargs):
+                out_kwargs = {}
+                for key, value in kwargs.items():
+                    if value is not None:
+                        out_kwargs[key] = value
+                return out_kwargs
+
+            def fit(self, Y, T, X=None, W=None, nuisances=None, *, sample_weight=None, sample_var=None):
+                Y_pred, = nuisances
+                if (X is not None) and (self._featurizer is not None):
+                    X = self._featurizer.fit_transform(X)
+                filtered_kwargs = _filter_none_kwargs(sample_weight=sample_weight, sample_var=sample_var)
+                if self._multitask_model_final:

```

Comment:
Could the same result be achieved by wrapping the model with the MultitaskModelWrapper?

Reply:
Good idea. tried it but failed. StatsModelsLinearRegression does not have get and set_params and hence cannot be wrapped by sklearn multioutputregressor. So until we make statsModelsLinearRegression comply with sklearn estimators (which could just be as simple as making the parent of that class be sklearn's BaseEstimator. But I'll leave it for a future PR. Please create an issue.

Reply:
I actually just also tried this but still it fails. So now StatsModelsLinearRegression is a BaseEstimator and can be wrapped by MultiOutputRegressor. But the problem is that I was allowing for StatsModelsLinearRegression to take as input X=None and this is not allowed by the MutliOutputRegressor. This then needs a lot of code refactoring, which I don't think is worth.

Reply:
Ok, thanks for giving it a shot.  Can you add a short comment to that effect to the code so that it's easier to know why this approach was taken?

Reply:
Done. Comment added.